### DISCUSSION TITLE
Expected Skript language

### DISCUSSION THREAD

**Discussion by dflvunoooooo** [joshuar/go-hass-agent#132] [Q&A] [answered]
What language does go-hass-agent expect the skripts to be written in? Is it only bash or is python or something else allowed as well?

**Comment by joshuar** [comment, marked as answer]
Any scripting language that supports being called via a [shebang](https://linuxhandbook.com/shebang/) line can be used. So Bash, Fish, Zsh, Python, Perl, Ruby possibly many more 🙂

The script feature won't work for anything that would require being compiled, so you cannot write scripts in Go for example. However, as the script feature is essentially just executing the named file. You could write your own Go program, compile it, then specify the resulting binary as a script.

As long as the output meets the [requirements](https://github.com/joshuar/go-hass-agent/blob/main/docs/scripts.md#output-format) in the documentation, it'll result in sensors being created in Home Assistant.

**Reply by dflvunoooooo** [reply]
Great, thank you for your answer!
